ui/src/api.ts

```diff
@@ -57,12 +57,18 @@ const api = {
         await setupAPI();
       }
 
+      client.onError = () => {
+        (async () => {
+          useLocalState.setState({ errorCount: errorCount + 1 });
+        })();
+      };
```

Comment:
This and the one in thread can safely be moved up into the `setupApi` call where we set the client. We don't need to set a new onError each time we poke/thread.